### DISCUSSION TITLE
Finish and return solution after 3,4 same solution

### DISCUSSION THREAD

**Discussion by SruthyJohn** [google/or-tools#3342] [Routing (and legacy CP) questions]
Hi Team,
I am using OR tools for finding the best delivery path between a huge number of stores. I am able to get the results for various datasets. But my concern is after a vague analysis of the data we are allotting searchParameters.TimeLimit as 2,5,10 mins etc. But it is noticed that sometimes we are getting same solution at 1,2,5,10 mins. So is there any option to return the feasible solution, if it is same for 3,4 times? 
From forums, I get to know that there is AddAtSolutionCallback by which we can achieve this but couldn't get much sample for that esp with C#.net. Can somebody please help on this?

**Comment by Mizux** [comment]
tracked by:
* https://github.com/google/or-tools/issues/3300
note: currently no ETA but I hope to work on it once release v9.4 is out...
